protocol/messenger_backup_handler.go

```diff
@@ -0,0 +1,205 @@
+		bh.messagesToProceed = nil
+		bh.lastKnownTime = message.Clock
+	}
+
+	bh.messagesToProceed = append(bh.messagesToProceed, message)
+}
+
+func (m *Messenger) startWaitingForTheLatestBackedupMessageLoop() {
+	ticker := time.NewTicker(timeToPostponeBackedUpMessagesHandling)
+	m.backedupMessagesHandler.postponeTasksWaitGroup.Add(1)
+	go func() {
+		defer m.backedupMessagesHandler.postponeTasksWaitGroup.Done()
+		for {
+			select {
+			case <-ticker.C:
+				if !m.backedupMessagesHandler.postponeHandling {
+					return
+				}
+				m.backedupMessagesHandler.postponeHandling = false
+
+				for _, msg := range m.backedupMessagesHandler.messagesToProceed {
+					messageState := m.buildMessageState()
+					errors := m.HandleBackup(messageState, msg)
+					if len(errors) > 0 {
+						for _, err := range errors {
+							m.logger.Warn("failed to handle postponed Backup messages", zap.Error(err))
+						}
+					}
+				}
+
+				m.backedupMessagesHandler.messagesToProceed = nil
+			case <-m.quit:
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+}
+
+func (m *Messenger) HandleBackup(state *ReceivedMessageState, message protobuf.Backup) []error {
+	var errors []error
+	if m.backedupMessagesHandler.postponeHandling {
+		m.backedupMessagesHandler.addMessage(message)
+		return errors
+	}
+
+	err := m.handleBackedUpProfile(message.Profile, message.Clock)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	for _, contact := range message.Contacts {
+		err = m.HandleSyncInstallationContact(state, *contact)
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	for _, community := range message.Communities {
+		err = m.handleSyncCommunity(state, *community)
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+	err = m.handleBackedUpSettings(message.Setting)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	if m.config.messengerSignalsHandler != nil {
+		response := wakusync.WakuBackedUpDataResponse{}
+
+		response.AddFetchingBackedUpDataDetails(SyncWakuSectionKeyProfile, message.ProfileDetails)
+		response.AddFetchingBackedUpDataDetails(SyncWakuSectionKeyContacts, message.ContactsDetails)
+		response.AddFetchingBackedUpDataDetails(SyncWakuSectionKeyCommunities, message.CommunitiesDetails)
+		response.AddFetchingBackedUpDataDetails(SyncWakuSectionKeySettings, message.SettingsDetails)
+
+		m.config.messengerSignalsHandler.SendWakuFetchingBackupProgress(&response)
+	}
+
+	return errors
+}
+
+func (m *Messenger) handleBackedUpProfile(message *protobuf.BackedUpProfile, backupTime uint64) error {
+	if message == nil {
+		return nil
+	}
+
+	dbDisplayNameClock, err := m.settings.GetSettingLastSynced(settings.DisplayName)
+	if err != nil {
+		return err
+	}
+
+	response := wakusync.WakuBackedUpDataResponse{
+		Profile: &wakusync.BackedUpProfile{},
+	}
+
+	if dbDisplayNameClock < message.DisplayNameClock {
+		err := m.SetDisplayName(message.DisplayName, false)
+		response.AddDisplayName(message.DisplayName, err == nil)
```

Comment:
we want to stop immediately at errors, better to fail fast, unless you expect an error, or you don't care about it, this does not seem to be the case

Reply:
What was my logic, if backup is made of multiple messages, where any of them may refer to more than a single thing, for example a single message may carry information about up to 20 contacts, profile message carries info about display name + identity images, why would we get back immediately instead of trying to apply as much as we can from the backup?

Reply:
I see why you'd want to do that, but it's better to throw red flags at this stage, we need to know why things don't work at early stages of development, otherwise we risk sweeping errors under the carpet, https://en.wikipedia.org/wiki/Fail-fast at this stage is better (at the message-level, we don't want to completely stop processing messages), since if anything throws an error here, it would be completely unexpected, but maybe there's a good reason